### DISCUSSION TITLE
Libraries repo

### DISCUSSION THREAD

**Discussion by ivelina-yordanova** [b2ihealthcare/snow-owl#619] [Q&A] [answered]
Hi,

This might have been asked already but couldn't find any information online or in the repo as to why there is no recent versions released [here](https://mvnrepository.com/artifact/com.b2international.snowowl).

Is there a way to pull the latest libs from somewhere? We are currently working around that by just building them ourselves but we are trying to move away from that option and wondering if there is something I missed and there is actually a way to get them from a different repo maybe ?

Thanks and regards,
Ivelina

**Comment by cmark** [comment]
Hi @ivelina-yordanova,

Those packages have been uploaded by the maintainer of this fork: https://github.com/IHTSDO/snow-owl.
That version of Snow Owl has been altered slightly compared to our vanilla version and those packages have been used by them mainly.
Unfortunately, we do not maintain any publicly available artifact repositories at the moment from where you could download Snow Owl's libraries.
However, we have been thinking a few months now to make them publicly available in the Maven Central repository or using [GitHub packages](https://github.com/b2ihealthcare/snow-owl/packages). There are two problems with either of these, they do not support p2 dependencies directly and the current pom.xml's used for the various packages does not come with dependencies section, since they are mainly OSGi bundles and not actual Maven modules. At the moment, we are not familiar with any solution that could resolve at least one of them.

For this reason, it is very likely that we will make them available in a custom Nexus repository hosted by us. We expect this to be resolved in the coming months.

Let me know if you have any further questions.

Cheers,
Mark

**Comment by cmark** [comment, marked as answer]
Hi @ivelina-yordanova,

In the last couple of weeks, we have been working hard to get our custom Maven Releases/Snapshots repositories up and running.
We are happy to announce that `snow-owl` and other selected open-source repositories of ours will be publicly available at the following repositories:
* Releases: https://nexus.b2i.sg/repository/maven-releases/
* Snapshots: https://nexus.b2i.sg/repository/maven-snapshots/

We will push the latest Snow Owl 7.16.2 version to the releases repository sometime later today. In the meantime, please use the `7.17.0-SNAPSHOT` version, if you would like to give it a shot.

Also, if you are interested a few SNOMED CT language-specific modules have been extracted to dedicated Git repositories for easier access:
* ECL - https://github.com/b2ihealthcare/snomed-ecl
* QL - https://github.com/b2ihealthcare/snomed-ql
* SCG - https://github.com/b2ihealthcare/snomed-scg
* ETL - https://github.com/b2ihealthcare/snomed-etl
However, this should not affect the consumption of the Snow Owl Terminology Server.

If you have any further questions, let me know.

Regards,
Mark
